client.go

```diff
@@ -718,23 +718,36 @@ func (c *Client[TTx]) Start(ctx context.Context) error {
 			producer := producer
 
 			if err := producer.StartWorkContext(fetchCtx, workCtx); err != nil {
-				stopProducers()
+				startstop.StopAllParallel(producersAsServices())
 				stopServicesOnError()
 				return err
 			}
 		}
 
 		return nil
 	}(); err != nil {
-		defer close(stopped)
+		defer stopped()
 		if errors.Is(context.Cause(fetchCtx), startstop.ErrStop) {
 			return rivercommon.ErrShutdown
 		}
 		return err
 	}
 
 	go func() {
-		defer close(stopped)
+		// Wait for all subservices to start up before signaling our own start.
+		// This isn't strictly needed, but gives tests a way to fully confirm
+		// that all goroutines for subservices are spun up before continuing.
+		//
+		// Stops also cancel the "started" channel, so in case of a context
```

Comment:
At first I thought there was a typo, but maybe it's just worded awkwardly. This feels a bit clearer imo, up to you:

```suggestion
		// Stop also cancels the "started" channel, so in case of a context
```